### PR TITLE
fixed compare_searches.feature

### PR DESCRIPTION
**Pull request by AndreaOlivieri**
The following commits fix a lot of problems with search.
- I added ling_0_id, ling_1_id in different commits for retrieve the ling ids and use them for assert some test steps.
- I enable the ability to make search request
- I deleted 'flush_result_groups' because doesn't work
- I change the use of FactoryGirl because it didn't work
- I fixed the last scenario in compare_search.feature because there is a part of that scenario it doesn't work

### CODE REVIEW

**Comment by dej611** [comment]
From a first review of this PR it smells a bit. Will spend a bit of time one of these days to review and test it better.

**Comment by AndreaOlivieri** [comment]
This is the hardest PR that I made and I think there is something has to be changed, but at least it works. Without these changes the compare search doesn't work at all

File: app/assets/javascripts/views/searches/results/results_default.js [hunk 1]
```diff
@@ -53,18 +59,31 @@
...
         'value_0'   : getValue('parent'),
         'example_0' : getExamples('parent'),
+        'ling_1_id' : getLingID('child'),
         'ling_1'    : getLing('child'),
         'property_1': getProperty('child'),
         'value_1'   : getValue('child'),
         'example_1' : getExamples('child')
       };
 
+      includeLingIds(columns);
```
Comment (AndreaOlivieri) [inline app/assets/javascripts/views/searches/results/results_default.js:87]: This is the point that I include ling ids to column result. After that it possible to extract the lings ids for the feature tests

File: app/assets/templates/searches/results/regular_results.hamstache [hunk 2]
```diff
@@ -27,7 +27,7 @@
       {{/header.example_1}}
   %tbody
     {{#rows}}
-    %tr
+    %tr.search_result{"data-parent-value" => "{{ling_0_id}}", "data-child-value" => "{{ling_1_id}}"}
```
Comment (AndreaOlivieri) [inline app/assets/templates/searches/results/regular_results.hamstache:30]: This is the point that ling ids are used.

File: app/controllers/searches_controller.rb [hunk 3]
```diff
@@ -156,9 +156,10 @@ def rescue_from_search_error(exception)
   def perform_search(offset=0)
     Search.new do |s|
       s.creator = current_user
-      s.group = current_group
-      s.query = params[:search]
-      s.offset = offset
+      s.group         = current_group
+      s.query         = params[:search]
+      s.result_groups = params[:result_groups]
```
Comment (AndreaOlivieri) [inline app/controllers/searches_controller.rb:163]: the only way to calculate a compare search is passing the result group param, without it compare search will calculate a wrong result.

File: features/compare_searches.feature [hunk 4]
```diff
@@ -141,8 +141,11 @@ Feature: Compare searches
     And I select "Second search" from "with"
     And I press "Go"
     Then I should see 3 search result rows
+    When I follow "Save Search"
```
Comment (AndreaOlivieri) [inline features/compare_searches.feature:145]: Save search link need to be clicked if you want to see the modal.

File: app/assets/javascripts/views/searches/results/results_default.js (hunk 1 above)
Comment (dej611) [reply app/assets/javascripts/views/searches/results/results_default.js:87]: Can you leave this comment in the code?

File: app/controllers/searches_controller.rb (hunk 3 above)
Comment (dej611) [reply app/controllers/searches_controller.rb:163]: Ditto

File: features/compare_searches.feature (hunk 4 above)
Comment (dej611) [reply features/compare_searches.feature:145]: Ditto

**Pull request merged**